src/cis/fam_cis_direct.cpp

```diff
@@ -351,11 +401,35 @@ void Fam_CIS_Direct::resize_region(uint64_t regionId, size_t nbytes,
         THROW_ERRNO_MSG(CIS_Exception, REGION_RESIZE_NOT_PERMITTED,
                         message.str().c_str());
     }
+    used_memsrv_cnt = region.used_memsrv_cnt;
+    std::list<std::shared_future<void>> resultList;
+    size_t bytes_per_server = nbytes / used_memsrv_cnt;
+    for (uint32_t id = 0; id < used_memsrv_cnt; id++) {
+        if (id == (used_memsrv_cnt - 1)) {
```

Comment:
We need to resize each memory server by equal quantity. If there is a remainder it has to be added to all memory servers.